checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInference.java

```diff
@@ -368,41 +442,77 @@ private void analyzeOwnershipTransferAtAssignment(
       if (!TreeUtils.isConstructor(methodTree)) {
         disposedFields.remove((VariableElement) lhsElement);
       }
-      addOwningToParamsIfDisposedAtAssignment(obligations, rhsObligation, rhs);
+
+      int paramIndex = getIndexOfParam(rhsObligation);
+      if (paramIndex == -1) {
+        return;
```

Comment:
I'm concerned about this early return, because it means that the "otherwise" part of the javadoc doesn't actually happen if `rhsObligation` isn't associated with a parameter. I'm struggling to articulate a situation in which that's incorrect, but it doesn't seem to preserve the same semantics here.

Reply:
I think it's fine because we are only tracking formal parameter aliases, and if the rhsObligation is not an alias of any of the formal parameters, it won't exist in the obligations set. So skipping `mcca.removeObligationsContainingVar(obligations, (LocalVariableNode) rhs);` line doesn't break anything.

Reply:
Can you add a comment just above the `return;` with this information?